### PR TITLE
Fix dependencies for Tesseract (fix issue #259)

### PR DESCRIPTION
**Pull request by stweil**
- Remove development packages for graphic libraries which are not required
- Add missing development packages
- Add required package for training tools

Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -658,9 +658,13 @@ ifneq ($(findstring tesseract, $(OCRD_MODULES)),)
 # Tesseract.
 
 # when not installing via PPA, we must cope without ocrd_tesserocr's deps-ubuntu-modules
-CUSTOM_DEPS += g++ make automake libleptonica-dev
-# but since we are building statically, we need more (static) libs at build time
```
Comment (stweil) [inline Makefile:662]: We disable building a shared Tesseract library, but we don't build statically. All shared libraries for the supported graphic formats should be installed as dependencies of `libleptonica-dev`.

Comment (bertsky) [reply Makefile:662]: > We disable building a shared Tesseract library, but we don't build statically. 

You are right: With `enable_shared=no enable_static=yes` I get …

    libtool: link: g++ -g -Og -O0 -DDEBUG -std=c++17 -o tesseract src/api/tesseract-tesseractmain.o  ./.libs/libtesseract.a -L/usr/local/lib /usr/local/lib/liblept.so -lpng16 -ljpeg -lgif -ltiff -lwebpmux -lwebp -lopenjp2 -lm -lz -larchive /usr/lib/x86_64-linux-gnu/libcurl-nss.so -lrt -lpthread -pthread

…which does link the binary dynamically.

But somehow I distinctly remember having to add `libgif-dev libjpeg-dev libpng-dev libtiff-dev` when we moved towards `--disable-shared` on your recommendation here.

Is it possible that in the meantime something changed in Tesseract build itself (`enable_shared_with_static_runtimes=no` perhaps)?

Comment (stweil) [reply Makefile:662]: There was no such change on the Tesseract side. Tesseract builds for Windows require `libtiff-dev`, but not for Linux or MacOS builds.

Comment (bertsky) [reply Makefile:662]: Ah, found it! This is a leftover of `LDFLAGS='-static'`, which has been abandoned.

https://github.com/OCR-D/ocrd_all/commit/07ab53c0f1701a0f7aa28568ee1e90c1522e8e76

**Review by bertsky — APPROVED** [review]

**Pull request merged**